File: testdata/fixtures/php/CWE-1004.php
<?php
/* CWE-1004: Sensitive Cookie Without HttpOnly Flag.
 *
 * Without HttpOnly a cookie is readable from document.cookie, so any script
 * that runs on the page - injected, or arriving through a dependency - can
 * read the session and send it elsewhere. It is the flag that turns an XSS
 * into an account takeover, and setting it costs one argument.
 */

function login($session_id, $token)
{
    /* ruleid: php-cookie-without-httponly */
    setcookie('session', $session_id);
    /* ruleid: php-cookie-without-httponly */
    setcookie('session', $session_id, time() + 3600);
    /* ruleid: php-cookie-without-httponly */
    setcookie('session', $session_id, time() + 3600, '/', '', true, false);
    /* ruleid: php-cookie-without-httponly */
    setcookie('remember', $token, ['expires' => time() + 86400, 'secure' => true]);
    /* ruleid: php-cookie-without-httponly */
    setrawcookie('session', $session_id, time() + 3600, '/');

    /* ok: php-cookie-without-httponly */
    setcookie('session', $session_id, time() + 3600, '/', '', true, true);
    /* ok: php-cookie-without-httponly */
    setcookie('remember', $token, [
        'expires' => time() + 86400,
        'secure' => true,
        'httponly' => true,
        'samesite' => 'Lax',
    ]);
    /* ok: php-cookie-without-httponly */
    session_set_cookie_params(['httponly' => true, 'secure' => true]);
}

File: testdata/fixtures/kotlin/CWE-1004.kt
// CWE-1004: the session cookie is readable from JavaScript.
//
// Without HttpOnly one cross-site script anywhere on the origin reads the
// session cookie and posts it somewhere. The flag is what makes an XSS bug a
// defacement rather than an account takeover.
//
// This file sets the flag somewhere, so the rule's file-level half - a cookie
// added in a file that never mentions setHttpOnly - is switched off here, and
// what is left is the explicit refusal.
package fixture

import javax.servlet.http.Cookie
import javax.servlet.http.HttpServletResponse

class Session {

    fun start(response: HttpServletResponse, id: String) {
        val cookie = Cookie("session", id)
        // ruleid: cookie-missing-httponly
        cookie.setHttpOnly(false)
        response.addCookie(cookie)
    }

    fun startSafely(response: HttpServletResponse, id: String) {
        val cookie = Cookie("session", id)
        // ok: cookie-missing-httponly
        cookie.setHttpOnly(true)
        response.addCookie(cookie)
    }
}
